handlers/deployments.go

```diff
@@ -26,3 +28,50 @@ func DeploymentList(w http.ResponseWriter, r *http.Request) {
 
 	RespondWithJSON(w, http.StatusOK, serviceList)
 }
+
+// DeploymentsIstioValidations is the API handler to get istio validations of all deployments that returns
+// an IstioValidation object which contains for each object type
+// one validation per each detected object in the current cluster that have an error/warning in its configuration.
+func DeploymentsIstioValidations(w http.ResponseWriter, r *http.Request) {
```

Comment:
I think we don't need this endpoint anymore. In service details we have the sidecar boolean added into the services list endpoint. I guess we are following same pattern.
If I understood correctly, from deployment list we run for each deployment the sidecar pod presence checkers for all the pods of the deployment. Then we count how many pods passed the sidecar pod presence checkers and we can then add the pods with sidecar / total pods into deployment overview struct.